src/Enhavo/Bundle/MediaLibraryBundle/Form/Type/ItemType.php

```diff
@@ -43,7 +50,30 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 'edit_route' => 'enhavo_media_library_admin_tag_update',
                 'frame_key' => 'media_library_tags'
             ])
+            ->add('url', TextType::class, [
+                'label' => 'media_library.form.label.url',
+                'translation_domain' => 'EnhavoMediaLibraryBundle',
+                'mapped' => false,
+                'data' => '',
+            ])
         ;
+
+        $builder->addEventListener(FormEvents::PRE_SET_DATA, function (FormEvent $event) {
+            $form = $event->getForm();
+
+            /** @var ?Item $data */
+            $data = $event->getData();
+
+            if ($data && $data->getFile() && $data->getFile() instanceof FileInterface) {
```

Comment:
`PRE_SET_DATA` will always be called, so you can just add this and remove the previous one outside the event listener.

```
$form->add('url', TextType::class, [
    'label' => 'media_library.form.label.url',
    'translation_domain' => 'EnhavoMediaLibraryBundle',
    'mapped' => false,
    'data' => $data?->getFile() ? $this->themeUrlGenerator->generate($data->getFile()) : null,
]);
```